Assets/Scripts/InteractableController.cs

```diff
@@ -1,33 +1,105 @@
 using UnityEngine;
-using UnityEngine.InputSystem;
+using TMPro;
 
 public class InteractableController : MonoBehaviour
 {
+  // Prefab reference
+  public Canvas interactionTip;
+  public string interactionTipText;
+  public bool allowMultipleInteractions = false;
+  IrritationBar irritationBar;
+
   [System.Serializable]
   public struct IrritationReaction
   {
     public NonPlayerController nonPlayerController;
     public float irritationModifier;
   }
   public List<IrritationReaction> irritationReactionList = new List<IrritationReaction>();
+  Canvas interactionTipInstance;
+
+  NonPlayerController reactingNPC;
+
+  void Awake()
+  {
+    initInteractionTip();
+  }
 
-  // Sample trigger
-  private void OnTriggerEnter(Collider other)
+  void initInteractionTip()
   {
-    if (other.gameObject.CompareTag("Player"))
+    if (interactionTip != null)
     {
-      foreach (IrritationReaction irritation in irritationReactionList)
-      {
-        handleUpdateIrritation(irritation.nonPlayerController, irritation.irritationModifier);
-      }
+      // Set position of interaction tool tip (attempt here to take into account size of object + camera position)
+      Quaternion rotation = Camera.main.transform.rotation;
+      Vector3 position = gameObject.transform.position;
+      position.y = gameObject.transform.localScale.y + position.y;
+      position.x = gameObject.transform.localScale.x + position.x;
+      position.z = gameObject.transform.localScale.z + position.z;
+
+      // Create new instance of tool tip based on prefab
+      interactionTipInstance = Instantiate(interactionTip, position, rotation);
+      interactionTipInstance.gameObject.SetActive(false);
+
+      SetInteractionTipText();
+    }
+  }
+
+  // Set the instructional interaction tool tip text
+  void SetInteractionTipText()
+  {
+    TextMeshProUGUI tipDisplayText = interactionTipInstance.GetComponentInChildren<TMPro.TextMeshProUGUI>();
+    tipDisplayText.text = interactionTipText;
+  }
+
+  public void ShowInteractionTip(bool isVisible)
+  {
+    if (interactionTipInstance != null)
+    {
+      interactionTipInstance.gameObject.SetActive(isVisible);
     }
   }
 
-  // Uses BroadcastMessage to send irritation to specific NPC
-  private void handleUpdateIrritation(NonPlayerController nonPlayerController, float irritationModifier)
+  public void HandleInteraction()
   {
-    nonPlayerController.BroadcastMessage("UpdateIrritation", irritationModifier);
+    // Loop through each attached NPC irritation reaction
+    foreach (IrritationReaction irritation in irritationReactionList)
+    {
+      setIrritationScore(irritation);
+    }
+    // Set the game object to inactive & destroy the interaction tool tip
+    if (!allowMultipleInteractions)
+    {
+      gameObject.SetActive(false);
+      Destroy(interactionTipInstance.gameObject);
+    }
+  }
+
+  // UpdateIrritation triggered by InteractableController
+  void setIrritationScore(IrritationReaction irritation)
```

Comment:
opinion: this could live in the nonPlayerController

the signal i'm seeing is that we're grabbing a lot of references from that class and doing work on them. since that class has all the references we need already, we could reduce how much this class needs to know about that class.